### PR TITLE
New EncryptorSessionDataTranscoder

### PR DESCRIPTION
**Pull request by barrantesgerman**
New implementation of `SessionDataTranscoder` to encrypt the data.

```java
@Override
protected RequestResponseFactory createRequestResponseFactory() {
    SessionDataStorage sessionDataStorage = new CookieSessionDataStorage(getPippoSettings(), new EncryptorSessionDataTranscoder("PasswordPassword"));
    SessionManager sessionManager = new SessionManager(sessionDataStorage);

    return new SessionRequestResponseFactory(this, sessionManager);
}
```

### CODE REVIEW

**Comment by decebals** [comment]
I don't think that we need a new module. `pippo-session` is the perfect place to put the new `EncryptedSessionDataTranscoder`.
Allso, in the future please try to make few commits for a PR (for example in this PR is enough a single commit) because it's more clear when we look in the (git) log.
Also, in the new `EncryptedSessionDataTranscoder` (and in general) I prefer to not use telescopic constructors. Maybe fluent setter methods (that returns this) or a inner Builder class (build pattern) is better.

**Comment by barrantesgerman** [comment]
Sorry for the delay :worried: , I make the requested changes :smile: 

```java
    @Override
    protected RequestResponseFactory createRequestResponseFactory() {
        EncryptorSessionDataTranscoder transcoder = EncryptorSessionDataTranscoder.builder().secretKey("PasswordPassword").build();
        SessionDataStorage sessionDataStorage = new CookieSessionDataStorage(getPippoSettings(), transcoder);
        SessionManager sessionManager = new SessionManager(sessionDataStorage);

        return new SessionRequestResponseFactory(this, sessionManager);
    }
```

As always, any suggestions are welcome

File: pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/EncryptorSessionDataTranscoder.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    private final SessionDataTranscoder transcoder;
+
+    public EncryptorSessionDataTranscoder(String secretKey, String hmacSHA1Key, SessionDataTranscoder transcoder, Encryptor encryptor) {
+        this.secretKey = secretKey;
+        this.hmacSHA1Key = hmacSHA1Key;
+        this.transcoder = transcoder;
+        this.encryptor = encryptor;
+    }
+
+    public static Builder builder() {
```
Comment (decebals) [inline pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/EncryptorSessionDataTranscoder.java]: We don't need this method EncryptedSessionDataTranscoder.Builder is enough.

File: pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/EncryptorSessionDataTranscoder.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    public static class Builder {
+
+        private String secretKey;
+        private String hmacSHA1Key;
+        private Encryptor encryptor;
+        private SessionDataTranscoder transcoder;
+
+        Builder() {
```
Comment (decebals) [inline pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/EncryptorSessionDataTranscoder.java]: Remove this constructor. The code is:
```java
new EncryptedSessionDataTranscoder.Builder.secretKey(secretKey).build();
```

File: pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/EncryptorSessionDataTranscoder.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+ */
+package ro.pippo.session;
+
+import ro.pippo.core.PippoRuntimeException;
+import ro.pippo.core.util.CryptoUtils;
+
+/**
+ * @author Herman Barrantes
+ */
+public class EncryptorSessionDataTranscoder implements SessionDataTranscoder {
```
Comment (decebals) [inline pippo-session-parent/pippo-session/src/main/java/ro/pippo/session/EncryptorSessionDataTranscoder.java]: I think that `EncryptedSessionDataTranscoder` is a better name.

**Comment by barrantesgerman** [comment]
Applied all suggested changes :+1:

**Comment by decebals** [comment]
Thanks!

**Pull request merged**